src/executor/current_thread.rs

```diff
@@ -458,37 +464,17 @@ impl CurrentRunner {
     where F: FnOnce(&Self) -> R,
     {
         CURRENT.with(|current| {
-            if current.scheduler.get().is_null() {
-                Err(())
-            } else {
+            if current.entered.get() {
                 Ok(f(current))
+            } else {
+                Err(())
             }
         })
     }
 
-    /// Set the provided scheduler to the TLS slot for the duration of the
-    /// closure.
-    ///
-    /// `CurrentThread::execute` will access the CURRENT thread-local variable in
-    /// order to push a future into the scheduler. This requires a `&mut`
-    /// reference, introducing mutability hazards.
-    ///
-    /// Rust requires that `&mut` references are not aliases, i.e. there are
-    /// never two "live" mutable references to the same piece of data. In order
-    /// to store a `&mut` reference in a thread-local variable, we must ensure
-    /// that one can not access the scheduler anywhere else.
-    ///
-    /// To do this, we only allow access to the thread local variable from
-    /// within the closure passed to `set_scheduler`. This function also takes a
-    /// &mut reference to the scheduler, which is essentially holding a "lock"
-    /// on that reference, preventing any other location in the code from
-    /// also getting that &mut reference.
-    ///
-    /// When `set_scheduler` returns, the thread-local variable containing the
-    /// mut reference is set to null. This is done even if the closure panics.
-    ///
-    /// This reduces the odds of introducing pointer aliasing.
-    fn set_scheduler<F, R>(&self, scheduler: &mut Scheduler, f: F) -> R
```

Comment:
There are various comments in the file still referring to this function/it's implications.